cmd/infracost/output.go

```diff
@@ -149,25 +149,30 @@ func outputCmd(ctx *config.RunContext) *cobra.Command {
 					ui.PrintWarning(cmd.ErrOrStderr(), "The dashboard is part of Infracost's hosted services. Contact hello@infracost.io for help.")
 				}
 
-				projectContexts := make([]*config.ProjectContext, len(combined.Projects))
-				for i := range combined.Projects {
-					projectContexts[i] = config.EmptyProjectContext()
-				}
-				combinedRunIds := []string{}
-				for _, input := range inputs {
-					if id := input.Root.RunID; id != "" {
-						combinedRunIds = append(combinedRunIds, id)
+				if len(inputs) == 1 && inputs[0].Root.RunID != "" {
+					result := inputs[0].Root
+					combined.RunID, combined.ShareURL = result.RunID, result.ShareURL
+				} else {
+					projectContexts := make([]*config.ProjectContext, len(combined.Projects))
+					for i := range combined.Projects {
+						projectContexts[i] = config.EmptyProjectContext()
 					}
-				}
-				ctx.SetContextValue("runIds", combinedRunIds)
+					combinedRunIds := []string{}
+					for _, input := range inputs {
+						if id := input.Root.RunID; id != "" {
+							combinedRunIds = append(combinedRunIds, id)
+						}
+					}
+					ctx.SetContextValue("runIds", combinedRunIds)
 
-				dashboardClient := apiclient.NewDashboardAPIClient(ctx)
-				result, err := dashboardClient.AddRun(ctx, projectContexts, combined)
-				if err != nil {
-					log.Errorf("Error reporting run: %s", err)
-				}
+					dashboardClient := apiclient.NewDashboardAPIClient(ctx)
+					result, err := dashboardClient.AddRun(ctx, projectContexts, combined)
+					if err != nil {
+						log.Errorf("Error reporting run: %s", err)
+					}
 
-				combined.RunID, combined.ShareURL = result.RunID, result.ShareURL
+					combined.RunID, combined.ShareURL = result.RunID, result.ShareURL
+				}
```

Comment:
could we separate this into a function so that we could return early and/or remove to much nesting. I generally hate `if/else` and try and avoid if possible. e.g. something like: 

```go

combined.RunID, combined.ShareURL = combineRuns(input)

```

with

```go

func combineRuns(inputs []output.ReportInput) (int64, string) {
    if len(inputs) == 1 && inputs[0].Root.RunID != "" {
        result := inputs[0].Root
	return result.RunID, result.ShareURL
    }
    
    ...
}
```